mmv1/templates/terraform/pre_update/secret_manager_regional_secret.go.erb

```diff
@@ -0,0 +1,18 @@
+<%- # the license inside this block applies to this file
+	# Copyright 2024 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+// As the API expects only one of ttl or expireTime
+if d.HasChange("ttl") && !d.HasChange("expire_time") {
```

Comment:
Shouldn't there be an opposite condition here:
To remove ttl if expiry_time is introduced in the update.

Reply:
As the API expects only one of the `ttl` or `expireTime` field, I've added this condition for switching between ttl and expireTime field. I've confirmed this behaviour and I've also added respective test case for the same. This test case can be found here: https://github.com/GoogleCloudPlatform/magic-modules/pull/11678/files#diff-11e793af43830f42f81e70d049956097a6df58b14f018cb62e7f1727f2bcd211R407

Reply:
We should add a test case to prove it.

Reply:
Yeah I've already added the test case that handles the switching between these fields.

Reply:
Thanks.